src/Implement/UnitTestFileOperator.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <memory>
+#include <thread>
+#include <unistd.h>
+
+#include "OperateData.h"
+#include "FileOperator.h"
+
+namespace  {
+    const uint16_t test_port = 9000;
+    std::string temp_response = "";
+
+    void TestClient() {
+        int sockfd = 0;
+        sockfd = socket(AF_INET, SOCK_STREAM, 0);
+        struct sockaddr_in info;
+        bzero(&info, sizeof(info));
+        info.sin_family = PF_INET;
+        info.sin_addr.s_addr = inet_addr("127.0.0.1");
+        info.sin_port = htons(test_port);
+        int err = connect(sockfd, (struct sockaddr *)&info, sizeof(info));
+
+        char buffer[2048] = {};
+        size_t n_recv = recv(sockfd, buffer, sizeof(buffer), 0);
+
+        temp_response = buffer;
+
+        close(sockfd);
+    }
+}
+
+// test with a sudo string data
+TEST(FILEOPERATOR_TEST, Basic) {
```

Comment:
As previously mentioned, consider adding MockResponseHandler and test the FileOperator class with it. You can do this in another PR.

Reply:
CallingAPI() itself will call ResponseHandler to send the result back to the client, so not need to add MockResponseHandler.